test/e2e/setup.go

```diff
@@ -40,16 +40,11 @@ import (
 
 // TestOption holds the options for integration test.
 type TestOption struct {
-	useCache bool
-	useMemdb bool
+	UseCache bool
+	UseMemdb bool
 }
 
-var generateGolden bool
-
-func init() {
-	flag.BoolVar(
-		&generateGolden, "generate_golden", false, "generate golden files")
-}
+var GenerateGolden = os.Getenv("GENERATE_GOLDEN") == "true"
```

Comment:
Does this need to be unset on exit?

Reply:
I think setting the env in the code has no effect to the actual environment. The env is set in update_golden.sh and scoped in that file, so would have no effect outside.